### PR TITLE
Fixed error message outputs/handling for help commands in CLI

### PR DESCRIPTION
**Pull request by rodkeys**
Starting jsipfs without any commands crashes it and gives a slightly unclear error message from yargs.

I changed this to have yargs display the original help message on starting jsipfs with no commands.

Also there is a fix for the --help command messages not being displayed at all.

### CODE REVIEW

**Comment by diasdavid** [comment]
@rodkeys thank you! Great catch, mind updating the tests to reflect this?

File: src/cli/bin.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ updateNotifier({
 const cli = yargs
   .commandDir('commands')
   .demandCommand(1)
+  .fail((msg, err, yargs) => {
+    if (err) throw err // preserve stack
+    console.error(yargs.help())
+    process.exit(1)
+  })
+
```
Comment (nunofmn) [inline src/cli/bin.js]: The Travis tests are failing, because the linting is failing with this added empty line :)

File: src/cli/bin.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ updateNotifier({
 const cli = yargs
   .commandDir('commands')
   .demandCommand(1)
+  .fail((msg, err, yargs) => {
+    if (err) throw err // preserve stack
+    console.error(yargs.help())
```
Comment (nunofmn) [inline src/cli/bin.js]: Here maybe you could use the [showHelp()](https://github.com/yargs/yargs#showhelpconsolelevelerror) function instead.

File: src/cli/bin.js (hunk 1 above)
Comment (rodkeys) [inline src/cli/bin.js]: Thank you for catching this, I will update my code

**Pull request merged**